indigo/indigo/src/main/scala/indigo/shared/events/GlobalEvent.scala

```diff
@@ -299,6 +303,61 @@ trait NetworkReceiveEvent extends GlobalEvent
 
 /** Events relating to manipulating locally stored data
   */
+
+enum StorageActionType:
+  case Save, Load, Delete, Find
+
+sealed trait StorageEventError extends GlobalEventError:
+  /** The identifier of the storage item accessed. Either the index (an Int), or the key (a String)
+    */
+  val key: String | Int
```

Comment:
I'm not sure how I feel about this. There's nothing wrong with it technically, but it feels like we're leaking implementation details / JavaScript.

The alternative would be something like:

```scala
enum StorageKey:
  case Index(value: String), Key(value: String)
```

...but undeniably it amounts to the same thing...